pkg/frontend/querymiddleware/generic_query_cache_test.go

```diff
@@ -164,81 +165,114 @@ func testGenericQueryCacheRoundTrip(t *testing.T, newRoundTripper newGenericQuer
 	for testName, testData := range tests {
 		t.Run(testName, func(t *testing.T) {
 			for reqName, reqData := range requestTypes {
-				t.Run(reqName, func(t *testing.T) {
-					// Mock the limits.
-					limits := multiTenantMockLimits{
-						byTenant: map[string]mockLimits{
-							userID: {
-								resultsCacheTTLForCardinalityQuery: testData.cacheTTL,
-								resultsCacheTTLForLabelsQuery:      testData.cacheTTL,
+				for _, reqMethod := range []string{ /*http.MethodGet ,*/ http.MethodPost} {
```

Comment:
```suggestion
				for _, reqMethod := range []string{http.MethodGet, http.MethodPost} {
```

Reply:
Fixed in https://github.com/grafana/mimir/pull/5524/commits/2bae1d325ca4231116176e76b7a41ca0c2afe672